rpc_state_reader/src/lib.rs

```diff
@@ -382,95 +484,186 @@ mod tests {
     }
 
     #[test]
-    fn test_invoke_execute() {
-        let contract_address = Address(felt_str!(
-            "06fcccb8c9c5bc490600d0d3a95134d3b2aacec7461fc1930178215803fa8d0c",
-            16
-        ));
-        let entry_point_selector = EXECUTE_ENTRY_POINT_SELECTOR.clone();
+    fn test_get_transaction() {
+        let rpc_state = RpcState::new(
+            RpcChain::MainNet,
+            BlockValue::Tag(serde_json::to_value("latest").unwrap()),
+        );
+        let tx_hash = "06da92cfbdceac5e5e94a1f40772d6c79d34f011815606742658559ec77b6955";
 
-        let max_fee = 103000000000000;
-        let version = felt_str!("100000000000000000000000000000001", 16);
+        rpc_state.get_transaction(tx_hash);
+    }
 
-        let calldata = vec![
-            felt_str!("1", 16),
-            felt_str!(
-                "49d36570d4e46f48e99674bd3fcc84644ddd6b96f7c741b1562b82f9e004dc7",
-                16
-            ),
-            felt_str!(
-                "83afd3f4caedc6eebf44246fe54e38c95e3179a5ec9ea81740eca5b482d12e",
-                16
-            ),
-            felt_str!("0", 16),
-            felt_str!("3", 16),
-            felt_str!("3", 16),
-            felt_str!(
-                "54ae4dbc24999badc9a161f6f4b72156ca7da92da93046fcbb48680a324ac7c",
-                16
-            ),
-            felt_str!("26f00fdabd0000", 16),
-            felt_str!("0", 16),
-        ];
+    /// Tested with the following query to the Feeder Gateway API:
+    /// https://alpha4-2.starknet.io/feeder_gateway/get_transaction_trace?transactionHash=0x019feb888a2d53ffddb7a1750264640afab8e9c23119e648b5259f1b5e7d51bc
```

Comment:
Do we have a way to mock this? Or use a fake server? Otherwise, we may get a flakey test due to network issues, server-side problems, etc.

Reply:
Yes, we should do that in before the Feeder Gateway gets deprecated.